http-client/src/main/java/io/avaje/http/client/DHttpClientRequest.java

```diff
@@ -545,13 +555,20 @@ public <T> HttpResponse<T> handler(HttpResponse.BodyHandler<T> responseHandler)
     return response;
   }
 
-  /**
-   * Prepare and send the request but not performing afterResponse() handling.
-   */
+  /** Prepare and send the request but not performing afterResponse() handling. */
   private <T> HttpResponse<T> sendWith(HttpResponse.BodyHandler<T> responseHandler) {
     context.beforeRequest(this);
     addHeaders();
-    final HttpResponse<T> response = performSend(responseHandler);
+    final HttpResponse<T> response;
+    if (errorMapper != null) {
+      try {
+        response = performSend(responseHandler);
+      } catch (final HttpException e) {
+        throw errorMapper.apply(e);
```

Comment:
This `errorMapper.apply()` isn't where I was expecting it to be. 

I was thinking it would be part of `checkMaybeThrow()` ... so I'm pondering this. This does not look right in that we want to let the response be decoded if necessary and HttpException is thrown later?

I'll pull this and see how it works and why it's not what I was expecting - looks cool otherwise.

Reply:
This is so it'll work for connection exceptions

Reply:
Hmm, wait you might be right

Reply:
Yeah, we need the mapping to occur also in the checkResponse() and checkMaybeThrow().

See the additional commits at: https://github.com/avaje/avaje-http/pull/275